bodo/pandas/managers.py

```diff
@@ -195,6 +197,29 @@ def _collect(self):
         """
         Collect data from workers if needed.
         """
+        # Execute the plan if it's lazy
+        if self._plan is not None:
+            from bodo.ext import plan_optimizer
+
+            optimized_plan = plan_optimizer.py_optimize_plan(self._plan)
```

Comment:
Not immediately obvious to me why constructing a lazy manager with a plan requires the immediate execution of that plan.  Some documentation on when this is called and why it is required behavior at that point.

Reply:
Collect requires the execution of the plan to return the results. It's not constructing a lazy manager. See spawn project documentation for more details.